### PR TITLE
Add implementation to get short id from store.getId() using a flag

### LINKED ISSUE #26: Ability to get short id from store.getId() using a flag
We can introduce a flag to make it easy to get short id from `store.getId()`. Something like this:
```ts
function getId(short: boolean = false) {
  // By default it returns the regular result. But if `short = true` is passed it returns the initial 8 characters of the id. 
  ...
}
```
**Usage:**
Example: https://github.com/leapfrogtechnology/async-store/blob/e899695eedf9aa6aa6cbb256f0618591396c08da/examples/express-ts/src/logger.ts#L9

```diff
- return (store.getId() || '').substring(0, 8); 
+ return store.getId(true);
```

### PR DESCRIPTION
**Pull request by silwalanish**
Resolves #26 
- Add `short` flag in `getId` to get short unique domain id.
    * Domain Implementation: Return first 8 characters of unique domain id if `short` is set.

### CODE REVIEW

File: src/impl/domain.ts [hunk 1]
```diff
@@ -212,12 +212,24 @@ export function getActiveDomain(): StoreDomainInterface {
...
+ * ```
+ * store.getId(); // Returns full id
+ * store.getId(true); // Returns first 8 characters of id. Default false
+ * ```
+ *
+ * @param {boolean} short Set `true` to get short id. Default `false`
+ * @returns {(string | undefined)} Returns full id by default. Set `short` to true to get short id.
  */
-export function getId(): string | undefined {
+export function getId(short: boolean = false): string | undefined {
```
Comment (mesaugat) [inline src/impl/domain.ts]: IMO this is a bit confusing. Just looking at `getId(true)` doesn't tell you anything. I think we can be more explicit and leave `getId()` as it is. What we can do is create another function `getShortId()` to return the trimmed version of the active domain id.

Thoughts? @kabirbaidhya @silwalanish

Comment (mesaugat) [reply src/impl/domain.ts]: And we have missing tests.

Comment (silwalanish) [reply src/impl/domain.ts]: Okay. I will add `getShortId()` to return the trimmed version and leave `getId()` as it is. And the `test`

File: src/index.ts [hunk 2]
```diff
@@ -180,12 +180,19 @@ export function isInitialized(): boolean {
 }
 
 /**
- * Gets the unique domain id created for the current context / scope.
+ * Get's the unique domain id created for the current context / scope.
```
Comment (cham11ng) [inline src/index.ts]:
```suggestion
 * Gets the unique domain id created for the current context/scope.
```

File: src/impl/domain.ts (hunk 1 above)
Comment (kabirbaidhya) [reply src/impl/domain.ts]: Yes, sounds good. Let's do it.

File: src/impl/domain.ts [hunk 3]
```diff
@@ -212,12 +212,24 @@ export function getActiveDomain(): StoreDomainInterface {
...
   const activeDomain = getActiveDomain();
+  const id = activeDomain && activeDomain[ID_KEY];
+
+  if (short) {
```
Comment (kabirbaidhya) [inline src/impl/domain.ts]: Make sure you've covered all the cases in the tests.

**Comment by kabirbaidhya** [comment]
@silwalanish Please make the changes as per the comments and add tests then it should be mergable.

**Comment by silwalanish** [comment]
> @silwalanish Please make the changes as per the comments and add tests then it should be mergable.

Will do first thing tomorrow🙂

**Comment by kabirbaidhya** [comment]
@silwalanish Not urgent brother. Enjoy Dashain :wink:.

**Pull request closed without merging**